cts/cli/regression.dates.exp

```diff
@@ -369,6 +369,14 @@ Duration:  3 months
 Duration ends at: 2009-04-30 00:00:00Z
 =#=#=#= End test: 2009-01-31 + 3 Months - OK (0) =#=#=#=
 * Passed: iso8601               - 2009-01-31 + 3 Months
+=#=#=#= Begin test: Test correct immediate plus-sign at timezone parsing =#=#=#=
+Date: 1556246574
+=#=#=#= End test: Test correct immediate plus-sign at timezone parsing - OK (0) =#=#=#=
+* Passed: iso8601               - Test correct immediate plus-sign at timezone parsing
+=#=#=#= Begin test: Like the previous, with exotic timezone to avoid false negative =#=#=#=
+Date: 1556246574
+=#=#=#= End test: Like the previous, with exotic timezone to avoid false negative - OK (0) =#=#=#=
+* Passed: iso8601               - Like the previous, with exotic timezone to avoid false negative
```

Comment:
@waltdisgrace Tests are failing because of this:
```
[2025-03-17T20:03:55.904Z]  =#=#=#= Begin test: Test correct immediate plus-sign at timezone parsing =#=#=#=
[2025-03-17T20:03:55.904Z] -Date: 1556246574
[2025-03-17T20:03:55.904Z] +Date: 1556242974
```

The difference between those two numbers is 3600 seconds, which is one hour, which is way too round a number to be a coincidence.  When I run it here, I get the same number as CI does.  How did you generate the new test output?  Do you perhaps have your timezone set to something unusual?

Reply:
We recently switched to daylight saving time. I would not be surprised if that's playing a role somehow. For example, if the system's timezone has an effect and if one of you has your system set to UTC and the other has your system set to local time.

Reply:
I don't think system time zone is supposed to play a role here though? Maybe there's some sort of rounding behavior that's implementation-dependent and differs on your two machines?

Reply:
Besides, shouldn't the resulting time be 2h20m later in the second test?

Reply:
There's a bug in our parsing. It only works if there's a space before the `+` or `-` sign.
```
$ iso8601 -d '2019-04-25T21:42:54+0200' -S
Date: 1556253774
$ iso8601 -d '2019-04-25T21:42:54 +0200' -S
Date: 1556221374
```

Reply:
@waltdisgrace Yeah, looks like our parsing code is wrong.  If you're curious, you can get a free version of the ISO8601 standard here:

https://www.loc.gov/standards/datetime/iso-tc154-wg5_n0038_iso_wd_8601-1_2016-02-16.pdf

Section 3.4.1 is pretty explicit that spaces aren't allowed in the time strings, and section 4.2.5.2 restates that again for the time offset portion.  So, the fact that we do not parse the timezone if there's a space means we're not handling the standard correctly.

This might be more than you were looking to get into, but it would be worth taking a look and seeing if you could fix that bug.  Of course, nothing is easy and we'll need to support our buggy behavior for now as well.

Reply:
> the fact that we do not parse the timezone if there's a space means we're not handling the standard correctly.

The fact that we don't parse the timezone UNLESS there's a space means we're not handling the standard correctly.
* Standard: Timezone is valid only WITH NO space.
* Pacemaker: Timezone is valid only WITH a space, and is ignored WITH NO space.

Reply:
Right, yes.  That's what I meant to say.  One day I'll learn to stop typing comments and talking in slack and talking in irc at the same time.